ipatests/test_integration/test_trust.py

```diff
@@ -223,6 +224,37 @@ def test_user_without_posix_attributes_not_visible(self):
         # Getent exits with 2 for non-existent user
         assert result.returncode == 2
 
+    def test_override_homedir(self):
+        """Check that POSIX attributes, such as shell or
+        home directory should not be overwritten or missing.
+
+        Related ticket https://pagure.io/SSSD/sssd/issue/2474
+        """
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        testuser = 'testuser@%s' % self.ad_domain
+        try:
+            domain = self.master.domain
+            tasks.modify_sssd_conf(
+                self.master,
+                domain.name,
+                {
+                    'subdomain_homedir': '%o'
+                }
+            )
+            tasks.clear_sssd_cache(self.master)
+            self.master.run_command(['getent', 'initgroups', '-s', 'sss', testuser])
+
+            result = self.master.run_command(['getent', 'passwd', testuser])
+            assert "/home/testuser" in result.stdout_text
+            log_file = "/var/log/sssd/sssd_{0}.log" .format(self.master.domain.name)
+            sssd_debug_log = self.master.get_file_contents(
+                log_file, encoding='utf-8'
+            )
+            assert 'get_subdomain_homedir_of_user failed' not in sssd_debug_log
```

Comment:
There is little problem in this check: the log file could contain this string BEFORE the testscase is executed. So we need to check that this line is not found in the part of log file which was written during execution of this testcase.
To do it you need to check the size of the log file before the test and then skip the initial part.
You could do it right in the testcase like
```
logsize = len(self.master.get_file_contents(log_file))
....
log = self.master.get_file_contents(log_file)[logsize:]
assert '...' not in log
```
but It would be nice to create utility in tasks.py, something like:
```
class LogFile(object):
    def __init__(self, host, path):
        self.host = host
        self.path = path
        self._initial_pos = len(host.get_file_contents(path))

    def new_text(self, encoding=None):
        s = self.host.get_file_contents(self.path)[self._initial_pos:]
        if encoding is not None:
            s = s.decode(encoding)
        return s
```
If you use this snippet without changes, please ask some else to review it.